proposals/interface-types/Explainer.md

```diff
@@ -386,10 +410,10 @@ the given name and signature (in the example, `log_ : [i32,i32] → []`).
 This example uses the `memory-to-string` instruction that was previously shown
 in an export adapter functions; now `memory-to-string` is lifting the argument
 of a `call-import` instead of lifting the return value of a `call-export`.
-Note also that `memory-to-string` has no `free` immediate in this example. While
-it would be *possible* to do so, since the caller of the adapter is wasm code,
-it's simpler and potentially more efficient to let the caller worry about when
-to free the string.
```

Comment:
This is one of the weakest aspects of C's memory model: lack of structure around who is responsible for free-ing allocated memory. It has to be done by application convention; but that is incompatible with generating adapter code automatically. (Our adapter instructions should not be C or C++ specific.)

Reply:
unique_ptr and shared_ptr might be able to do some of this automatically, but yeah in general this will need to be done manually.

Reply:
Agreed.  Anything to change here or can I resolve this thread?

Reply:
Also keep in mind that there are use cases where you might not want to dynamically allocate/free anything; eg, maybe you're calling `log` with a constant string, or a string generated on your linear memory stack.

In practice, the host should worry about dynamic allocation when passing data to/from the callee, not the caller.

Reply:
Yes, but such cases are handled by simply not using *any* `defer-call-export` (as in the walkthrough's first section).